src/Workspaces/Core/Portable/Utilities/AbstractSpeculationAnalyzer.cs

```diff
@@ -521,6 +526,40 @@ private bool ReplacementChangesSemanticsForNode(SyntaxNode currentOriginalNode,
             return false;
         }
 
+        /// <summary>
+        /// Determine if removing the cast could cause the semantics of System.Object method call.
+        /// </summary>
+        private bool ReplacementBreaksSystemObjectMethodResolution(SyntaxNode currentOriginalNode, SyntaxNode currentReplacedNode, SyntaxNode previousOriginalNode, SyntaxNode previousReplacedNode)
+        {
+            if (previousOriginalNode != null && previousReplacedNode != null)
+            {
+                var originalExpressionSymbol = this.OriginalSemanticModel.GetSymbolInfo(currentOriginalNode).Symbol;
+                var replacedExpressionSymbol = this.SpeculativeSemanticModel.GetSymbolInfo(currentReplacedNode).Symbol;
+
+                if (IsExpressionSymbolSystemObjectMethod(originalExpressionSymbol) && IsExpressionSymbolSystemObjectMethod(replacedExpressionSymbol))
+                {
+                    var previousOriginalType = this.OriginalSemanticModel.GetTypeInfo(previousOriginalNode).Type;
+                    var previousReplacedType = this.SpeculativeSemanticModel.GetTypeInfo(previousReplacedNode).Type;
+
+                    return !Equals(previousOriginalType, previousReplacedType);
+                }
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Determines if the symbol is a non-overridable, non static method on System.Object (e.g. GetType)
+        /// </summary>
+        private bool IsExpressionSymbolSystemObjectMethod(ISymbol symbol)
```

Comment:
Not sure why "Expression" is inside of the name here. Just call this IsSystemObjectInstanceMethod?